test/utils/bytecode.hpp

```diff
@@ -181,6 +181,81 @@ inline bytecode sload(bytecode index)
     return index + OP_SLOAD;
 }
 
+template <evmc_opcode kind>
+struct call_instruction
+{
+private:
+    bytecode m_address = 0;
+    bytecode m_gas = 0;
+    bytecode m_value = 0;
+    bytecode m_input = 0;
+    bytecode m_input_size = 0;
+    bytecode m_output = 0;
+    bytecode m_output_size = 0;
+
+public:
+    explicit call_instruction(bytecode address) : m_address{std::move(address)} {}
+
+    auto& gas(bytecode g)
```

Comment:
Seems it would be more logical and better constraining the caller, if you had `uint64_t` parameter here (and in other methods), and `m_gas = bytecode{g};` in the body.

Reply:
I don't want to complicate this. This is rather a tool for presenting code.

Reply:
Also you can do `.gas(calldataload(0))`.